### PR TITLE
feat: geo 46 create api to provide freida with suppressed data and calculated table data

### PR DESCRIPTION
**Pull request by goemen**
# Description

The data has to be sent de-normalized and should include the Report, Company, Naics Code, Employee Range, Calculated and Suppressed Data.

Fixes # ([issue](https://finrms.atlassian.net/browse/GEO-422))

## Type of change
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- [ ] I have setup swagger ui in the backend external api, open `https://pay-transparency-pr-315-backend-external.apps.silver.devops.gov.bc.ca/api/v1/docs/`
- [ ]  Click the authorize button in the UI and enter your development api-key
- [ ] Test the api endpoint and verify that data returned has the correct shape

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---
Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://pay-transparency-pr-315-frontend.apps.silver.devops.gov.bc.ca)

---
Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://pay-transparency-pr-315-frontend.apps.silver.devops.gov.bc.ca)

### CODE REVIEW

File: backend-external/src/app.ts [hunk 1]
```diff
@@ -70,9 +71,18 @@ app.get(
       logger.error(`Health check failed: ${e}`);
       res.status(500).send('Health check failed');
     }
-  })
+  }),
 );
 
+if (process.env.NODE_ENV !== 'production') {
```
Comment (sukanya-rath) [inline backend-external/src/app.ts:77]: lets remove the if clause, will always display swagger

File: backend-external/src/utils.ts [hunk 2]
```diff
@@ -1,10 +1,31 @@
...
+    openapi: '3.1.0',
+    info: {
+      title: 'Pay Transparency External API',
+      version: '1.0.0',
+      description:
+        'These api exposes endpoint to query pay transparency reports',
+    },
+    servers: [
+      {
+        url: 'http://localhost:3002/api/v1/pay-transparency',
```
Comment (sukanya-rath) [inline backend-external/src/utils.ts:22]: can we pass env var here, and replace it with actual route url from helm charts something like `URL`
refer here https://github.com/bcgov/fin-pay-transparency/blob/main/charts/fin-pay-transparency/charts/backend-external/templates/route.yaml#L12

File: backend/src/v1/routes/external-consumer-routes.ts [hunk 3]
```diff
@@ -5,19 +5,28 @@ import { codeService } from '../services/code-service';
...
-    const endDate = req.query.endDate.toString();
-    const offset = Number(req.query.offset.toString());
-    const limit = Number(req.query.limit.toString());
-    if(isNaN(offset) || isNaN(limit)) {
-      return res.status(400).json({ error: 'Invalid offset or limit' });
+router.get(
+  '/',
+  utils.asyncHandler(async (req: Request, res: Response) => {
+    try {
+      console.log(req.query)
```
Comment (sukanya-rath) [inline backend/src/v1/routes/external-consumer-routes.ts:12]: make it log.silly

File: backend/src/v1/services/external-consumer-service.ts [hunk 4]
```diff
@@ -34,10 +40,92 @@ const externalConsumerService = {
     }
     // TODO: Add logic to fetch data from prismaReadOnlyReplica, below query needs to be updated.
```
Comment (sukanya-rath) [inline backend/src/v1/services/external-consumer-service.ts:41]: remove the TODO comments

**Review by sukanya-rath — CHANGES_REQUESTED** [review]

File: backend-external/src/utils.ts (hunk 2 above)
Comment (goemen) [reply backend-external/src/utils.ts:22]: I will need your help with this on Monday @sukanya-rath

File: backend-external/src/config/index.ts [hunk 5]
```diff
@@ -16,10 +16,11 @@ config.defaults({
       windowMs: process.env.RATE_LIMIT_WINDOW_MS || 60000, // 1 minute
       limit: process.env.RATE_LIMIT_LIMIT || 100, // Limit each IP to 100 requests per `window` (here, per 1 minute)
     },
+    url: process.env.EXTERNAL_BACKEND_SERVER_URL || 'http://localhost:3002'
   },
   backend:{
     apiKey: process.env.BACKEND_EXTERNAL_API_KEY || 'api-key',
-    url: process.env.BACKEND_URL || 'http://localhost:3000'
+    url: process.env.BACKEND_URL || 'http://localhost:3010'
```
Comment (sukanya-rath) [inline backend-external/src/config/index.ts:23]: Any particular reasons we need to make it 3010?

File: charts/fin-pay-transparency/charts/backend-external/templates/deployment.yaml [hunk 6]
```diff
@@ -42,6 +42,8 @@ spec:
               value: '1000'
             - name: LOG_LEVEL
               value: {{.Values.app.env.logLevel}}
+            - name: EXTERNAL_BACKEND_SERVER_URL
```
Comment (sukanya-rath) [inline charts/fin-pay-transparency/charts/backend-external/templates/deployment.yaml:45]: Rename to BASE_URL

**Review by sukanya-rath — CHANGES_REQUESTED** [review]

File: backend-external/src/config/index.ts (hunk 5 above)
Comment (goemen) [reply backend-external/src/config/index.ts:23]: The external api app in the backend is running on port 3010

File: charts/fin-pay-transparency/charts/backend-external/templates/deployment.yaml (hunk 6 above)
Comment (goemen) [reply charts/fin-pay-transparency/charts/backend-external/templates/deployment.yaml:45]: Do you mean just `BASE_URL` or `EXTERNAL_BACKEND_SERVER_BASE_URL `? @sukanya-rath

Comment (sukanya-rath) [reply charts/fin-pay-transparency/charts/backend-external/templates/deployment.yaml:45]: BASE_URL

**Pull request merged**